packages/go_router/lib/src/route.dart

```diff
@@ -1127,7 +1127,18 @@ class StatefulNavigationShell extends StatefulWidget {
       /// Recursively traverses the routes of the provided StackedShellBranch to
       /// find the first GoRoute, from which a full path will be derived.
       final GoRoute route = branch.defaultRoute!;
-      return _router.configuration.locationForRoute(route)!;
+      final List<String> parameters = <String>[];
+      patternToRegExp(route.path, parameters);
+      assert(
```

Comment:
Add test for this assertion case too?

Reply:
It looks like the route constructor already asserted this. so this assert is not reachable. I removed the comment since and treated it as a sanity check since it is no longer user facing check